peps/pep-0750.rst

```diff
@@ -809,79 +796,172 @@ At the time of this PEP's announcement, a fully-working implementation is
+source text:
 
-No Implicit String Concatenation
---------------------------------
+- ``Interpolation.format_spec`` defaults to ``""`` if not provided. It is therefore
+  impossible to distinguish ``t"{expr}"`` from ``t"{expr:}"``.
+- The debug specifier, ``=``, is treated as a special case. It is therefore not
+  possible to distinguish ``t"{expr=}"`` from ``t"expr={expr}"``.
+- Finally, format specifiers in f-strings allow arbitrary nesting. In this PEP 
+  and in the reference implementation, the specifier is eagerly evaluated 
+  to set the ``format_spec`` in the ``Interpolation``, thereby losing 
+  the original expressions. For example:
 
-Implicit tag string concatenation isn't supported, which is `unlike other string literals
+.. code-block:: python
+
+    spec = ".2f"
+    value = 42
+    template = t"{value:{spec}}"
+    assert template.args[1].format_spec == ".2f"
+
+We do not anticipate that these limitations will be a significant issue in practice.
+Developers who need to obtain the original template string literal can always
+use ``inspect.getsource()`` or similar tools.
+
+
+String Concatenation
+--------------------
+
+Implicit template string concatenation isn't supported, which is `unlike all other string literals
 <https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation>`_.
 
 The expectation is that triple quoting is sufficient. If implicit string
-concatenation is supported, results from tag evaluations would need to
+concatenation is supported, results from template evaluations would need to
 support the ``+`` operator with ``__add__`` and ``__radd__``.
 
-Because tag strings target embedded DSLs, this complexity introduces other
+Because template strings target embedded DSLs, this complexity introduces other
 issues, such as determining appropriate separators. This seems unnecessarily
 complicated and is thus rejected.
 
+
 Arbitrary Conversion Values
 ---------------------------
 
 Python allows only ``r``, ``s``, or ``a`` as possible conversion type values.
 Trying to assign a different value results in ``SyntaxError``.
 
-In theory, tag functions could choose to handle other conversion types. But this
+In theory, template functions could choose to handle other conversion types. But this
 PEP adheres closely to :pep:`701`. Any changes to allowed values should be in a
 separate PEP.
 
+
+Mechanism to Describe the "Kind" of Template
+--------------------------------------------
+
+If t-strings prove popular, it may be useful to have a way to describe the
+"kind" of content found in a template string: "sql", "html", "css", etc. 
+This could enable powerful new features in tools such as linters, formatters, 
+type checkers, and IDEs. (Imagine, for example, ``black`` formatting HTML in 
+t-strings, or ``mypy`` checking whether a given attribute is valid for an HTML 
+tag.) While exciting, this PEP does not propose any specific mechanism. It is 
```

Comment:
Let's beef up Koudai's acknowledgements: PEP review, evaluation sandboxes, etc.

Reply:
@pauleveritt I took a stab at this but please take another!